File: ios-app/BoostT1D/generate_app_icons.py
#!/usr/bin/env python3

from PIL import Image, ImageDraw
import math

def draw_heart(draw, center_x, center_y, size, color):
    """Draw a heart shape"""
    # Scale factor for the heart
    scale = size * 0.35
    
    # Heart shape using curves
    points = []
    for i in range(360):
        t = math.radians(i)
        # Parametric heart equation
        x = 16 * math.sin(t) ** 3
        y = -(13 * math.cos(t) - 5 * math.cos(2*t) - 2 * math.cos(3*t) - math.cos(4*t))
        
        # Scale and translate
        px = center_x + x * scale / 16
        py = center_y + y * scale / 16
        points.append((px, py))
    
    draw.polygon(points, fill=color)

def draw_blood_drop(draw, center_x, center_y, size, color):
    """Draw a blood drop shape"""
    # Water drop shape
    points = []
    for i in range(180):
        t = math.radians(i)
        x = center_x + size * 0.05 + math.sin(t) * size * 0.08
        y = center_y - size * 0.1 + i / 180 * size * 0.15
        points.append((x, y))
    
    # Complete the drop with a curve back to start
    for i in range(180, 0, -1):
        t = math.radians(i)
        x = center_x + size * 0.05 - math.sin(t) * size * 0.08
        y = center_y - size * 0.1 + i / 180 * size * 0.15
        points.append((x, y))
    
    draw.polygon(points, fill=color)

def create_app_icon(size):
    """Create app icon at specified size"""
    # Create image with blue background circle
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Blue circle background
    blue = (0, 122, 255)  # iOS blue
    draw.ellipse([0, 0, size, size], fill=blue)
    
    # White heart in center
    white = (255, 255, 255)
    heart_size = size * 0.5
    draw_heart(draw, size // 2, size // 2, heart_size, white)
    
    # Red blood drop on heart
    red = (255, 59, 48)  # iOS red
    drop_size = size * 0.2
    draw_blood_drop(draw, size // 2, size // 2, drop_size, red)
    
    return img

# Icon sizes needed for iOS
icon_sizes = [
    ("20", 20),
    ("29", 29),
    ("40", 40),
    ("58", 58),
    ("60", 60),
    ("76", 76),
    ("80", 80),
    ("87", 87),
    ("120", 120),
    ("152", 152),
    ("167", 167),
    ("180", 180),
    ("1024", 1024)
]

print("🎨 Generating app icons...")

for name, size in icon_sizes:
    icon = create_app_icon(size)
    filename = f"AppIcon-{name}x{name}.png"
    icon.save(filename)
    print(f"✅ Generated: {filename}")

print("\n🎉 All app icons generated successfully!")
print("\nNext steps:")
print("1. Open Xcode")
print("2. Navigate to BoostT1D/Assets.xcassets/AppIcon")
print("3. Drag and drop the generated PNG files into the corresponding slots")
print("4. Build and run to see your new app icon!")

File: ios-app/BoostT1D/generate_app_icons.swift
#!/usr/bin/env swift

import SwiftUI
import UIKit

// AppIconView definition
struct AppIconView: View {
    let size: CGFloat

    var body: some View {
        ZStack {
            // Blue circle background
            Circle()
                .fill(Color.blue)
                .frame(width: size, height: size)

            // White heart cutout
            AppIconHeartShape()
                .fill(Color.white)
                .frame(width: size * 0.7, height: size * 0.7)

            // Red blood drop on heart
            AppIconBloodDropShape()
                .fill(Color.red)
                .frame(width: size * 0.15, height: size * 0.2)
                .offset(x: size * 0.05, y: -size * 0.1)
        }
    }
}

struct AppIconHeartShape: Shape {
    func path(in rect: CGRect) -> Path {
        var path = Path()
        
        let width = rect.width
        let height = rect.height
        let centerX = width / 2
        let centerY = height / 2
        
        // Create a proper heart shape using two circles and a triangle
        // Left circle of heart
        let leftCircleCenter = CGPoint(x: centerX - width * 0.15, y: centerY - height * 0.05)
        let leftCircleRadius = width * 0.2
        
        // Right circle of heart  
        let rightCircleCenter = CGPoint(x: centerX + width * 0.15, y: centerY - height * 0.05)
        let rightCircleRadius = width * 0.2
        
        // Start from the bottom point of the heart
        path.move(to: CGPoint(x: centerX, y: centerY + height * 0.25))
        
        // Left curve up to the left circle
        path.addQuadCurve(
            to: CGPoint(x: leftCircleCenter.x - leftCircleRadius, y: leftCircleCenter.y),
            control: CGPoint(x: centerX - width * 0.1, y: centerY + height * 0.1)
        )
        
        // Left circle arc
        path.addArc(
            center: leftCircleCenter,
            radius: leftCircleRadius,
            startAngle: .degrees(180),
            endAngle: .degrees(0),
            clockwise: false
        )
        
        // Right circle arc
        path.addArc(
            center: rightCircleCenter,
            radius: rightCircleRadius,
            startAngle: .degrees(180),
            endAngle: .degrees(0),
            clockwise: false
        )
        
        // Right curve down to the bottom
        path.addQuadCurve(
            to: CGPoint(x: centerX, y: centerY + height * 0.25),
            control: CGPoint(x: centerX + width * 0.1, y: centerY + height * 0.1)
        )
        
        path.closeSubpath()
        
        return path
    }
}

struct AppIconBloodDropShape: Shape {
    func path(in rect: CGRect) -> Path {
        var path = Path()

        let width = rect.width
        let height = rect.height

        path.move(to: CGPoint(x: width / 2, y: 0))
        path.addQuadCurve(to: CGPoint(x: width / 2, y: height),
                          control: CGPoint(x: width, y: height * 0.75))
        path.addQuadCurve(to: CGPoint(x: width / 2, y: 0),
                          control: CGPoint(x: 0, y: height * 0.75))
        return path
    }
}

// Icon sizes needed
let iconSizes: [(String, CGFloat)] = [
    ("20x20", 20),
    ("29x29", 29),
    ("40x40", 40),
    ("58x58", 58),
    ("60x60", 60),
    ("76x76", 76),
    ("80x80", 80),
    ("87x87", 87),
    ("120x120", 120),
    ("152x152", 152),
    ("167x167", 167),
    ("180x180", 180),
    ("1024x1024", 1024)
]

// Generate icons
for (name, size) in iconSizes {
    let iconView = AppIconView(size: size)
    let renderer = ImageRenderer(content: iconView)
    renderer.scale = 1.0
    
    if let uiImage = renderer.uiImage,
       let pngData = uiImage.pngData() {
        let filename = "AppIcon-\(name).png"
        let url = URL(fileURLWithPath: FileManager.default.currentDirectoryPath).appendingPathComponent(filename)
        try? pngData.write(to: url)
        print("Generated: \(filename)")
    }
}

print("✅ All app icons generated!")



